crates/ethereum/payload/src/lib.rs

```diff
@@ -351,7 +351,21 @@ where
             )
             .map_err(|err| PayloadBuilderError::Internal(err.into()))?;
 
-        Some(Requests::new(vec![deposit_requests, withdrawal_requests, consolidation_requests]))
+        let mut requests = Requests::default();
+
+        if !deposit_requests.is_empty() {
+            requests.push_request(core::iter::once(0).chain(deposit_requests).collect());
+        }
+
+        if !withdrawal_requests.is_empty() {
+            requests.push_request(core::iter::once(1).chain(withdrawal_requests).collect());
+        }
+
+        if !consolidation_requests.is_empty() {
+            requests.push_request(core::iter::once(2).chain(consolidation_requests).collect());
+        }
```

Comment:
we probably want this functionality directly on the `Requests` type?

can leave it in this pr, but worth upstreaming imo